server/runtime/src/main/java/org/infinispan/server/configuration/ServerConfigurationBuilder.java

```diff
@@ -54,6 +57,27 @@ public EndpointsConfigurationBuilder endpoints() {
    @Override
    public void validate() {
       Arrays.asList(interfaces, socketBindings, security, endpoints).forEach(Builder::validate);
+      // When authz is enabled, ensure that at least one endpoint has authn, otherwise the server will be useless
+      if (builder.security().authorization().isEnabled()) {
```

Comment:
shouldn't this code be in `org.infinispan.server.configuration.endpoint.EndpointsConfigurationBuilder#validate()`?

Reply:
@tristantarrant ^

Reply:
I also do feel it would be better in `EndpointConfigurationBuilder`. We do instanceof checks for these two classes. Trying to keep those in the same approximate place would be a lot better.